src/libespeak-ng/synthdata.c

```diff
@@ -81,7 +81,7 @@ static espeak_ng_STATUS ReadPhFile(void **ptr, const char *fname, int *size, esp
 		fclose(f_in);
 		return ENOMEM;
 	}
-	if (fread(*ptr, 1, length, f_in) != length) {
+	if (fread(*ptr, 1, length, f_in) != (size_t)length/*>=0*/) {
```

Comment:
Line 71 changes for length < 0, but the comparison of a (size_t) result from fread with an (int) variable quite rightly produces a warning from GCC - a negative (int) will be treated as a large positive number (very large in my case because size_t is 64-bit).  The cast quiets the warning, but casts are evil... the comment explains why it is safe.  I sort-of gave up at this point - there are way too many warnings even without -Wall, -Wextra.  This is what the FEXTRA change is above - see the comments in the earlier commits.

Reply:
if (fread(*ptr, 1, length, f_in) != (size_t)length/*>=0; checked above*/) {

Reply:
The cast is fine, I'm just questioning the usefulness of the comment.

Reply:
I used a "SAFE" comment, IRC there are some commenting styles used by GCC and the various semi-commercial security checking programs.  Anyway, it's another convention; a programmer can check a piece of code and assert that even though it looks dubious it is, actually, safe.  The convention I changed it to follows the one in libpng, and libpng was being run through one of the semi-commercial code checkers.